tests/decorators/test_async.py

```diff
@@ -118,3 +118,34 @@ async def expensive_function(**kwargs: Any) -> int:
         assert await expensive_function(a="a") == 1
 
     m_set.assert_called_with(mock.ANY, mock.ANY, time_to_live=timedelta(seconds=42), if_not_exists=mock.ANY)
+
+
+def sync_exclude(key_, value_):
+    return int(key_) + value_ < 40
+
+
+async def async_exclude(key_, value_):
+    return sync_exclude(key_, value_)
+
+
+@pytest.mark.parametrize(
+    "exclude_function",
+    [
+        pytest.param(sync_exclude, id="sync-callable"),
+        pytest.param(async_exclude, id="async-callable"),
+    ],
+)
+async def test_exclude(cache: Cache[int, int], exclude_function):
+    @cached(
+        cache,
+        make_key=lambda _, arg: str(arg),
+        exclude=exclude_function,
+    )
+    async def power_function(arg: int) -> int:
+        return arg**2
+
+    with mock.patch.object(cache, "set", wraps=cache.set) as m_set:
+        assert await power_function(5) == 25
+        assert await power_function(6) == 36
+
+    m_set.assert_called_once_with("6", 36, time_to_live=mock.ANY, if_not_exists=mock.ANY)
```

Comment:
Picky/optional: it'd be easier to just check if the value was cached via `.get()`

Reply:
yess the less mocks the better :)